### PR TITLE
Permit easily purging unmanaged repos

### PR DESCRIPTION
**Pull request by jcpunk**
#### Pull Request (PR) description
With the support for defining repos, it is helpful to have an easy way to purge unmanaged repos.

#### This Pull Request (PR) fixes the following issues
N/A

### CODE REVIEW

File: manifests/init.pp [hunk 1]
```diff
@@ -104,13 +110,24 @@
   Array[String] $repo_exclusions = [],
   Hash[String, Hash[String, String]] $gpgkeys = {},
   String $utils_package_name = 'yum-utils',
+  Boolean $purge_unmanaged_repos = false,
+  Stdlib::Unixpath $repodir,
```
Comment (bastelfreak) [inline manifests/init.pp:114]: please set the default value here, not in hiera. This enables puppet-strings to pick up the value for the REFERENCE.md

Comment (jcpunk) [reply manifests/init.pp]: Done

**Review by bastelfreak — APPROVED** [review]

**Comment by jcpunk** [comment]
Can you release a new forge version with the last batch of changes?

**Comment by mphilipps** [comment]
@jcpunk does this actually work? If I just create a purge file resource for the directory, the module creates the repos and the purges them again each run. Apparently the modules creates files without using the file resources.

**Comment by mphilipps** [comment]
```
Jul  4 13:46:10 puppettest-linux-centos-7-64bit puppet-agent[11376]: (/Stage[main]/Yum/Yumrepo[extras]/ensure) created (corrective)
...
Jul  4 13:46:18 puppettest-linux-centos-7-64bit puppet-agent[11376]: (/Stage[main]/Yum/File[/etc/yum.repos.d/extras.repo]/ensure) removed (corrective)
```
That is from one run.

**Comment by jcpunk** [comment]
I'm a bit surprised that the `yumrepo` type doesn't provide the file resources....

My target use case is paired with https://github.com/voxpupuli/puppet-rhsm/commit/1c2abe59cc8b3459dc95f4ad6c85db5a51d2fbdb so that I can be sure only `redhat.repo` is in `/etc/yum.repos.d/` on certain systems.

**Comment by kenyon** [comment]
This was an attempted fix of #94.

**Pull request merged**